ncm-mysql/src/main/pan/components/mysql/schema.pan

```diff
@@ -34,10 +34,19 @@ function component_mysql_check_db_script = {
   };
 };
 
+# The function validating password imposes a non-empty string.
+function component_mysql_password_valid = {
+  if (  match(SELF, '^[^\\]+$') ) {
+    true;
+  } else {
+    false;
+  };
+};
+
 type component_mysql_user_right = string with match(SELF, '^(ALL( PRIVILEGES)?|ALTER( ROUTINE)?|CREATE( (ROUTINE|TEMPORARY TABLES|USER|VIEW))?|DELETE|DROP|EVENT|EXECUTE|FILE|GRANT OPTION|INDEX|INSERT|LOCK TABLES|PROCESS|REFERENCES|RELOAD|REPLICATION (CLIENT|SLAVE)|SELECT|SHOW (DATABASES|VIEW)|SHUTDOWN|SUPER|TRIGGER|UPDATE|USAGE)$');
 
 type component_mysql_db_user = {
-  'password'  : string with match(SELF, '^[^\\]+$')
+  'password'  : string with (length(SELF) == 0) || component_mysql_password_valid(SELF)
```

Comment:
The idea is that validation function consider that a valid password is not empty and for some property we allow it to be either empty or valid... this is the idea.